### PR TITLE
Fix broken drops

### PR DESCRIPTION
**Pull request by SlyryD**
- There is a deku shield drop from a pot in the Spirit Temple child side Anubis room that does not appear in the vanilla game, and logic might require you to get a deku shield this way.
- There is a magic jar on top of the Gerudo Training Ground eye statue that does not always refill your magic in the vanilla game.

### CODE REVIEW

**Comment by r0bd0g** [comment]
It might be in your wheelhouse so I figured I'd point out that there are 2 recovery hearts in both MQ DC and MQ BotW that were probably intended to be visible?

**Comment by r0bd0g** [comment]
I noticed you didn't add the shield pot to glitched world? I don't know enough about glitched to speculate why not. You would expect the logic to be the same as getting to the previous chest (plus the drops setting) unless the shield only spawns as child or something.

File: data/LogicHelpers.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     # "Bow": "'Bow'",
     # "Slingshot": "'Slingshot'",
     "Bombs": "Bomb_Bag",
-    "Deku_Shield": "Buy_Deku_Shield",
+    "Deku_Shield": "Buy_Deku_Shield or Deku_Shield_Drop",
```
Comment (cjohnson57) [inline data/LogicHelpers.json:23]: Are we sure that this overrides the normal behavior wherein the item "Deku Shield" (i.e. a normal non repeatable one) would be used where `Deku_Shield` is referenced?

Comment (fenhl) [reply data/LogicHelpers.json:23]: That seems like the sort of thing that would have been noticed by now if it was broken.

Comment (cjohnson57) [reply data/LogicHelpers.json:23]: Well before now it was using `Buy_Deku_Shield` directly

Comment (fenhl) [reply data/LogicHelpers.json:23]: Ah, I didn't realize that. Still, the `Goron_Tunic` helper is in use, and I'm positive that one works as intended.

Comment (cjohnson57) [reply data/LogicHelpers.json:23]: Good point, I think we can probably call this safe then.

**Pull request merged**